VirtualEconomyFramework/VENFTApp-Blazor/wwwroot/jsInterop.js

```diff
@@ -292,91 +312,117 @@ window.jsFunctions = {
+                        console.log(this.width + " " + this.height);
+
+                        
+                        var canvas = document.createElement("canvas");
+                        canvas.width = loadedImage.width;
+                        canvas.height = loadedImage.height;
+                        var ctx = canvas.getContext("2d");
+                        ctx.drawImage(loadedImage, 0, 0, loadedImage.width, loadedImage.height);
+
+                        var imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
                         var code = jsQR(imageData.data, imageData.width, imageData.height, {
                             inversionAttempts: "dontInvert",
-                        });
-
-                        if (code) {
-                            console.log("Found QR code", code.data);
-
-                            if (dotnetRef) {
-                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', code.data);
-                            }
-                        }
-                        else {
-                            if (dotnetRef) {
-                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', 'Cannot Load QR Code!');
+                        });
+
+                        if (code) {
+                            console.log("Found QR code", code.data);
+
+                            if (dotnetRef) {
+                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', code.data);
+                            }
+                        }
+                        else {
+                            if (dotnetRef) {
+                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', 'Cannot Load QR Code!');
                             }
                             console.log("No code found");
-                        }
-                    };
-                }*/
-            } catch (error) {
-                if (dotnetRef) {
-                    dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', `Cannot Load QR Code! ${error}`);
-                }
-            }
-
-        };
-        input.click();
-    },
-};
+                        }
+                    };
+                }*/
+            } catch (error) {
+                if (dotnetRef) {
+                    dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', `Cannot Load QR Code! ${error}`);
+                }
+            }
+
+        };
+        input.click();
+    },
+    modal: {
+        add: function (element) {
+            const container = document.getElementById("modal-container");
+            if (element && container) {
+                container.appendChild(element);
+            }
+        },
+        dispose: async function (element) {
+            if (!element || !element instanceof Element)
+                return;
+            handleModalClose(element);
+            element.remove();
+        },

```

Comment:
This is to make sure we remove modal element when dispose modal component. It's mainly for nested modals.